### ISSUE TITLE
Ec2 component support for windows

### ISSUE THREAD

**Issue by roleyfoley** [hamlet-io/engine-plugin-aws#299]
## Expected Behaviour

Using a deployment profile I can specify the image source and extensions required to deploy a microsoft windows based ec2 instance as an ec2 component

## Current Behaviour

While we support specifying your own image and extensions for compute instances a user would need to implement everything themselves without support from hamlet itself

## Possible Solution

- Add a new masterdata profile `_compute_windows2019` which is used to set the appropriate compute instance configuration for our recommended windows instance setup
- Add extensions which implement the compute tasks we include in the ec2 component based on windows requirements 

## Context

Adding support for this would highlight the hamlet flexibility in ec2 setup and how it works

**Comment by roleyfoley** [comment]
@ken9ross  This ticket is to cover off the basics of the windows support with a focus on the ec2 component

For the ec2 component we have the following tasks which I've divided into groups: 

base system configuration - these tasks don't have a dependency on other components or configuration and are mostly focussed on getting the system up and running 
```
                        COMPUTE_TASK_RUN_STARTUP_CONFIG,
                        COMPUTE_TASK_AWS_CFN_SIGNAL,
                        COMPUTE_TASK_AWS_CFN_WAIT_SIGNAL,
                        COMPUTE_TASK_OS_SECURITY_PATCHING,
                        COMPUTE_TASK_AWS_CLI
```

Extensions and profiles - these tasks use information collected from either extension/fragment processing or from reference data 
```
                       COMPUTE_TASK_SYSTEM_VOLUME_MOUNTING,
                       COMPUTE_TASK_HAMLET_ENVIRONMENT_VARIABLES,
                       COMPUTE_TASK_USER_BOOTSTRAP,
                       COMPUTE_TASK_SYSTEM_LOG_FORWARDING,
                       COMPUTE_TASK_FILE_DIR_CREATION
```

Link based config - these tasks use links from the component to other components to perform different tasks 
```
                        COMPUTE_TASK_DATA_VOLUME_MOUNTING,
                        COMPUTE_TASK_AWS_LB_REGISTRATION,
                        COMPUTE_TASK_EFS_MOUNT,
                        COMPUTE_TASK_USER_ACCESS,
````

From a complexity point of few they get more complicated as things get involved. The base system configuration tasks are used across most of our ec2 based components then as they get more complicated they change for each component

The ec2 component does have special configuration to handle some extra dependency requirements but its what would be the base level of configuration.

**Comment by roleyfoley** [comment]
Have a look through the current extensions in the aws provider which implement each of the tasks. That should give you an idea of the structure and how things are implemented. 

While developing the different extensions you can use a catch all extension which does an echo to stop exceptions from being logged during processing 

Something like 

```
[#ftl]

[@addExtension
    id="computetask_windows_notavailable"
    aliases=[
        "_computetask_windows_notavailable"
    ]
    description=[
       "Catch all for missing windows tasks"
    ]
    supportedTypes=[
        EC2_COMPONENT_TYPE,
        ECS_COMPONENT_TYPE,
        COMPUTECLUSTER_COMPONENT_TYPE,
        BASTION_COMPONENT_TYPE
    ]
    scopes=[
        COMPUTETASK_EXTENSION_SCOPE
    ]
/]

[#macro shared_extension_computetask_windows_notavailable_deployment_computetask occurrence ]

    [@computeTaskConfigSection
        computeTaskTypes=[ 
                       COMPUTE_TASK_SYSTEM_VOLUME_MOUNTING,
                       COMPUTE_TASK_HAMLET_ENVIRONMENT_VARIABLES,
                       COMPUTE_TASK_USER_BOOTSTRAP,
                       COMPUTE_TASK_SYSTEM_LOG_FORWARDING,
                       COMPUTE_TASK_FILE_DIR_CREATION,
                        COMPUTE_TASK_DATA_VOLUME_MOUNTING,
                        COMPUTE_TASK_AWS_LB_REGISTRATION,
                        COMPUTE_TASK_EFS_MOUNT,
                        COMPUTE_TASK_USER_ACCESS,
        ]
        id="NotAvailable"
        priority=1
        engine=AWS_EC2_CFN_INIT_COMPUTE_TASK_CONFIG_TYPE
        content={}
    /]

[/#macro]
```

Then as you add each task remove them from this list

**Comment by ml019** [comment]
@roleyfoley think this is done ?

**Comment by roleyfoley** [comment]
Yep this has been implemented in #301
